tests/testsupport/broker.py

```diff
@@ -53,7 +56,52 @@ def fake_broker():
     # print('Setup broker')
     broker = FakeBroker()
 
-    yield broker
+    try:
```

Comment:
I don't think that try-catch is necessary here. I think that py.test takes care of that.